### ISSUE TITLE
[Feature Request] kubeshark deploys pod on tainted nodes

### ISSUE THREAD

**Issue by mateuszdrab** [kubeshark/kubeshark#1318]
### Is your feature request related to a problem? Please describe.

I've just tried Kubeshark in my cluster and noticed that it was not able to start due to the following error:
```
ERR tapRunner.go:347 > Watching events. event=kubeshark-hub.174c6d65e4a8bdc3 kind=Pod name=kubeshark-hub note="Failed to pull image \"docker.io/kubeshark/hub:latest\": rpc error: code = NotFound desc = failed to pull and unpack image \"docker.io/kubeshark/hub:latest\": no match for platform in manifest: not found" pod=kubeshark-hub reason=Failed 
```

Quick check shows that the pods are scheduled to start on my 'special' worker nodes which are tainted and not available for standard deployments (the are RPIs with 32-bit Raspbian)

The above error is due to a missing build for the platform, which I don't mind; however, I don't see a way to override the scheduling behavior.

Nearly all my workload is running on amd64 nodes, so I was a bit surprised why kubeshark chose the tainted nodes for the deployment.

### Describe the solution you'd like to see

Based on my understanding, there is no armv7/v6 image available, so two solutions can be devised:
- schedule the pods on arm64/amd64 nodes automatically or provide a flag to control this
- use the K8s native deployment like a replication controller and let K8s handle the scheduling
- provide armv7/v6 images for those niche users (I'm not sure why those platforms are so neglected when majority of non enterprise users run a form of device reliant on it)

### Provide additional context

No extra context, just a thanks for creating this tool 😉

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mateuszdrab** [comment]
Any feedback on this?

**Comment by mertyildiran** [comment]
@mateuszdrab would you like to ignore the nodes with `effect: "NoSchedule"`?

**Comment by mateuszdrab** [comment]
> @mateuszdrab would you like to ignore the nodes with `effect: "NoSchedule"`?

Yes I think that would help, and/or perhaps create a deployment instead and let kubernetes decide where to launch the pod.

**Comment by mertyildiran** [comment]
@mateuszdrab implemented with https://github.com/kubeshark/kubeshark/commit/e4684a10afee4257d439f74013d37ab2d35a8525 please build it from source and try the new `--ignoreTainted` flag with:

```shell
git clone git@github.com:kubeshark/kubeshark.git && \
cd kubeshark/ && \
make && \
./bin/kubeshark__ tap -A --ignoreTainted
```

and please give the feedback of whether this new `--ignoreTainted` flag works for you or not.

**Comment by mateuszdrab** [comment]
> @mateuszdrab implemented with https://github.com/kubeshark/kubeshark/commit/e4684a10afee4257d439f74013d37ab2d35a8525 please build it from source and try the new `--ignoreTainted` flag with:
> 
> ```shell
> git clone git@github.com:kubeshark/kubeshark.git && \
> cd kubeshark/ && \
> make && \
> ./bin/kubeshark__ tap -A --ignoreTainted
> ```
> 
> and please give the feedback of whether this new `--ignoreTainted` flag works for you or not.

Thanks, I'll give it a try later.

I do wonder if perhaps a solution like I mentioned in comment https://github.com/kubeshark/kubeshark/issues/1284#issuecomment-1482000599 would be better though.

It would kill two birds with one stone.

**Comment by mateuszdrab** [comment]
@mertyildiran it is possible to override the tolerations of a DaemonSet, so I think the behavior could still be introduced - alternatively, node-selectors could be utilized as well to filter on nodes through a `--node-selectors` flag. 

I never got to the point of the daemonset deployment in my instance, since it never launched the hub and frontend pods - I guess that would be a new problem.

Theoretically, one might want to run the hub and frontend on a different type of node than where the ds would run... perhaps a separate set of configuration parameters would be an even better solution.

**Comment by mateuszdrab** [comment]
@mertyildiran I built the version from the post above and it's unforunately still trying to run the front and hub pods on my RPI nodes which are tainted `hardware=rpi:NoSchedule` and I ran this command line `./bin/kubeshark__ tap  -n haproxy --ignoreTainted`
